pip/req/req_install.py

```diff
@@ -343,11 +343,18 @@ def setup_py(self):
         assert self.source_dir, "No source dir for %s" % self
         try:
             import setuptools  # noqa
-        except ImportError:
+        except ImportError as e:
+            # Python 2 and 3 variants:
+            if e.args[0] in (
+                    "No module named setuptools",
+                    "No module named 'setuptools'"):
```

Comment:
Rather than poking at this, I suggest checking for 'setuptools' in sys.modules. Its a defined interface, the error message is not.